bindings/python/pymongoarrow/bson.pyi

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def process_bson_stream(bson_stream, context):
+    cdef const uint8_t* docstream = <const uint8_t *>bson_stream
+    cdef size_t length = <size_t>PyBytes_Size(bson_stream)
+    cdef bson_reader_t* stream_reader = bson_reader_new_from_data(docstream, length)
+    cdef cbool reached_eof = False
+    cdef const bson_t * doc = bson_reader_read(stream_reader, & reached_eof)
+    cdef bson_iter_t doc_iter
+    cdef const char* key
+    cdef bson_type_t value_t
+
+    builder_map = context.builder_map
+    type_map = context.type_map
+
+    while doc != NULL:
+        if not bson_iter_init(&doc_iter, doc):
+            raise RuntimeError
+        while bson_iter_next(&doc_iter):
+            key = bson_iter_key(&doc_iter)
+            if key in builder_map:
```

Comment:
What happens when a key is missing from one of docs? I think we should append a Null but it seems like we would skip it altogether. For example what happens in this scenario:
```
docs = [{'_id': 1, 'data': 10},
        {'_id': 2},
        {'_id': 3, 'data': 30},
        {'data': 40}]
schema = Schema({'_id': int32(), 'data': int64()})
...
```

Reply:
In this PR or a follow up PR?

Reply:
Done.